index.js

```diff
@@ -0,0 +1,311 @@
+        toDo.splice(index, 1);
+        taskField.removeChild(currentTask);
+        toDoCount--
+        document.getElementById('toDo-count').innerHTML = ' (' + toDoCount + ')';
+    } else {
+        completed.splice(index, 1);
+        completeField.removeChild(currentTask);
+        completedToDoCount--
+        document.getElementById('completed-count').innerHTML = ' (' + completedToDoCount + ')';
+    }
+    saveData();
+}
+
+function sortFromNew() {
+    toDo.sort(function(a, b) {
+        return new Date(a.date) - new Date(b.date);
+    });
+    taskField.innerHTML = "";
+    toDo.forEach(function(el) {
+        return createTaskField(el, taskField);
+    })
+    saveData();
+}
+
+function sortFromOld() {
+    toDo.sort(function(a, b) {
+        return new Date(b.date) - new Date(a.date);
+    });
+    taskField.innerHTML = "";
+    toDo.forEach(function(el) {
+        return createTaskField(el, taskField);
+    })
+    saveData();
+}
+
+function changeTheme() {
+    if (switcher.checked) {
+        document.querySelector('.navbar').classList.remove('bg-light');
+        document.querySelector('.modal-content').style.backgroundColor = '#0e1621';
+        document.body.style.backgroundColor = '#0e1621';
+        document.body.style.color = 'white';
+        switcher.nextElementSibling.textContent = 'Dark Mode';
+    } else {
+        document.querySelector('.navbar').classList.add('bg-light');
+        document.querySelector('.modal-content').style.backgroundColor = 'white';
+        document.body.style.backgroundColor = 'white';
+        document.body.style.color = 'black';
+        switcher.nextElementSibling.textContent = 'White Mode';
+    }
+}
+
+
+newTask.addEventListener('click', createNewTask)
+addTaskButton.addEventListener('click', addTask);
+editTaskButton.addEventListener('click', saveEditChanges);
+newSort.addEventListener('click', sortFromNew);
+oldSort.addEventListener('click', sortFromOld);
+switcher.addEventListener('click', changeTheme)
+
+
+function saveData() {
+    localStorage.setItem('toDo', JSON.stringify(toDo));
+    localStorage.setItem('completedToDo', JSON.stringify(completed));
+}
+
+
+window.onload = function loadTasks() {
+    if (localStorage.getItem('toDo') !== null) {
+        toDo = JSON.parse(localStorage.getItem('toDo'));
+        taskField.innerHTML = "";
+        toDoCount = toDo.length;
+        document.getElementById('toDo-count').innerText = ' (' + toDoCount + ')';
+        toDo.forEach(function(el) {
+            createTaskField(el, taskField);
+        })
+    }
+    if (localStorage.getItem('completedToDo') !== null) {
+        completed = JSON.parse(localStorage.getItem('completedToDo'));
+        completeField.innerHTML = "";
+        completedToDoCount = completed.length;
+        document.getElementById('completed-count').innerText = ' (' + completedToDoCount + ')';
+        completed.forEach(function(el) {
+            createTaskField(el, completeField);
+        })

```

Comment:
you repeat the same code twice, it differs only by `toDo` or `completedTodo`, you should create 1 method